### ISSUE TITLE
Add machine-readable JSON index of all files on download.mapsforge.org

### ISSUE THREAD

**Issue by mvl22** [mapsforge/mapsforge-creator#23]
It would be useful if a machine-readable index of all the files on download.mapsforge.org (and therefore any proxy cache that third-parties run locally to avoid traffic) could be created, so that apps wanting to show a list of available areas could easily ingest such an index.

Currently, downloads are produced and end up at:
http://download.mapsforge.org/maps/v5/
However, this gives an Apache directory listing over multiple URL locations, which is non-trivial to parse.

I realise this might be more a question of deployment for the administrators of the server, as `mapsforge-creator` seems to run on a per-file basis, but perhaps there is scope for achieving the above by converting the repetitive scripts at
https://github.com/mapsforge/mapsforge-creator/tree/master/v4
towards an architecture which uses a wrapper script to read from a JSON file of the same locations and run each invocation that way. That would then mean an index file essentially already exists (assuming no conversion failures).

**Comment by devemux86** [comment]
mapsforge-creator uses OpenStreetMap data extracts from [Geofabrik](https://download.geofabrik.de/) following its structure.

Mapsforge server is provided for everyone to download maps - not to be used directly in applications.
Developers wanting a downloader inside their apps, should have their own server hosting their maps.

Nevertheless I'll be happy to discuss improvements in the map creation process, keeping it in shell level for simple / easy maintenance. :slightly_smiling_face:

**Comment by mvl22** [comment]
> Mapsforge server is provided for everyone to download maps - not to be used directly in applications.
Developers wanting a downloader inside their apps, should have their own server hosting their maps.

Yes, naturally - as I implied, a proxy cache is what people should be doing.

But even with that, I still think it would be useful to have an index.

The result does indeed seem to match the Geofabrik structure, so I've tried to create an index by parsing theirs and rewriting things locally.
